### PR TITLE
Update EMR script with async bootstrapping

### PR DESCRIPTION
**Pull request by Xenorith**

### CODE REVIEW

File: integration/emr/alluxio-emr.sh [hunk 1]
```diff
@@ -237,32 +482,49 @@ If a different Alluxio version is desired, see the -d option.
...
   local root_ufs_uri="${1}"
+  # note that since the script args are shifted
+  # the shifted argument needs to be manually added when launching the background process
   shift
-  while getopts "b:cd:f:i:n:p:s:" option; do
-    if [[ -n "${OPTARG-}" ]]; then
-      OPTARG=$(echo -e "${OPTARG}" | tr -d '[:space:]')
-    fi
+
+  while getopts "ehb:cd:f:i:n:p:s:v:" option; do
```
Comment (LuQQiu) [inline integration/emr/alluxio-emr.sh:517]: -c                Install the alluxio client jars only.？
Where assign value to client_only?

File: integration/emr/alluxio-emr.sh [hunk 2]
```diff
@@ -280,179 +542,113 @@ USAGE_END
...
+  wait_for_hadoop "${is_master}" "${is_krb}"
 
+  echo "Starting Alluxio configuration"
+  set_custom_alluxio_properties "${delimited_properties}"
 
-  # Append default configs to site properties if the user hasn't set them already
+  # set auto generated properties
+  echo "Setting auto-generated properties in ${ALLUXIO_SITE_PROPERTIES}"
   doas alluxio "echo '# BEGIN AUTO-GENERATED PROPERTIES' >> ${ALLUXIO_SITE_PROPERTIES}"
   doas alluxio "echo '# Override these by specifying an alluxio-site.properties file, or delimited args within the EMR bootstrap' >> ${ALLUXIO_SITE_PROPERTIES}"
```
Comment (LuQQiu) [inline integration/emr/alluxio-emr.sh:640]: This conflicts with our alluxio replacement strategy, consider removing it?

File: integration/emr/alluxio-emr.sh (hunk 1 above)
Comment (Xenorith) [reply integration/emr/alluxio-emr.sh:517]: https://github.com/Alluxio/alluxio/pull/11545

File: integration/emr/alluxio-emr.sh (hunk 2 above)
Comment (Xenorith) [reply integration/emr/alluxio-emr.sh:640]: hmm yeah probably should remove this echo statement

File: integration/emr/alluxio-emr.sh (hunk 1 above)
Comment (Xenorith) [reply integration/emr/alluxio-emr.sh:517]: oh shit i didn't add it in here

Comment (Xenorith) [reply integration/emr/alluxio-emr.sh:517]: oh wait i dont need to cuz it was already committed...

Comment (LuQQiu) [reply integration/emr/alluxio-emr.sh:517]: i saw the line difference

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**